src/scripts/enhanced_lifestyle_analyses/enhanced_lifestyle_analyses.py

```diff
@@ -1,10 +1,834 @@
+                for gender, desc in self.gender_des.items():
+                    for cat in categories:
+                        gc_df[f'cat_{cat}'] = self.dfs[li_property][urban_rural][gender][cat].sum(axis=1)
+
+                    # normalise the probabilities
+                    gc_df = gc_df.apply(lambda row: row / row.sum(), axis=1)
+                    # do plotting
+                    ax = gc_df.plot(kind='bar', stacked=True, ax=axes[_cols_counter, _rows_counter],
+                                    title=f"{self._rural_urban_state[urban_rural]}"
+                                          f" {self.en_props[li_property][0]} categories in {desc}",
+                                    ylabel=f"{self.en_props[li_property][0]} proportions", xlabel="Year"
+                                    )
+                    self.custom_axis_formatter(gc_df, ax, li_property)
+                    # increase counter
+                    _rows_counter += 1
+
+                _cols_counter += 1
+            # save and display plots
+            add_footnote(f'{self.en_props[li_property][1]}')
+            plt.tight_layout()
+            plt.savefig(self.outputpath / (
+                li_property + '_' + self.datestamp + '.png'),
+                        format='png')
+            plt.show()
+
+        # 2. if property is not in a group of those that need plotting by urban and rural plot them by age category only
+        else:
+            if li_property == 'li_ed_lev':
+                # GENERAL EDUCATION PLOTTING
+                counter: int = 0  # counter for indexing purposes
+                # create subplots
+                fig, axes = plt.subplots(nrows=1, ncols=2, figsize=(10, 5))
+                for gender, desc in self.gender_des.items():
+                    # re-order columns
+                    re_ordered_df = self.dfs[li_property].reorder_levels([0, 2, 1, 3], axis=1)
+                    for cat in categories:
+                        gc_df[f'cat_{cat}'] = re_ordered_df[gender][cat].sum(axis=1)
+
+                    # normalise the probabilities
+                    gc_df = gc_df.apply(lambda row: row / row.sum(), axis=1)
+
+                    # do plotting
+                    ax = gc_df.plot(kind='bar', stacked=True, ax=axes[counter],
+                                    title=f"{desc} {self.en_props[li_property][0]}  categories",
+                                    ylabel=f"{self.en_props[li_property][0]} proportions", xlabel="Year"
+                                    )
+                    self.custom_axis_formatter(gc_df, ax, li_property)
+                    # increase counter
+                    counter += 1
+
+                # save and display plots for property categories by gender
+                add_footnote(f'{self.en_props[li_property][1]}')
+                plt.tight_layout()
+                plt.savefig(self.outputpath / (li_property + self.datestamp + '.png'), format='png')
```

Comment:
```suggestion
                fig.tight_layout()
                fig.savefig(self.outputpath / (li_property + self.datestamp + '.png'), format='png')
```
Same rationale as above